File: photostore-ionic2/src/app/photos/photo.ts
export interface Photo {
  id?: number;
  checksum?: string;
  gphotoId?: string;
  mimeType?: string;
  creationDate?: Date;
  filename?: string;
  thumbnailPhotoFile?: PhotoFile;
  photoFile?: PhotoFile;
}

export interface PhotoFile {
  base64URL?: string;
  url?: string;
  height: number;
  width: number;
}

export const getWidthForDesiredHeight = (maxHeight: number, photoFile: PhotoFile): number => {
  return Math.ceil((maxHeight / photoFile.height) * photoFile.width);
};

File: photostore_flutter/lib/models/photo.dart
import 'package:photostore_flutter/models/agnostic_media.dart';

class Photo extends AgnosticMedia {
  final String checksum;
  final String gphotoId;
  final String mimeType;
  final String filename;

  const Photo(
      {id,
      this.checksum,
      this.gphotoId,
      this.mimeType,
      creationDate,
      this.filename,
      thumbnail,
      thumbnailProvider,
      getThumbnailProviderOfSize})
      : super(
            id: id,
            creationDate: creationDate,
            thumbnail: thumbnail,
            thumbnailProvider: thumbnailProvider,
            getThumbnailProviderOfSize: getThumbnailProviderOfSize);

  @override
  List<Object> get props =>
      [...super.props, checksum, gphotoId, mimeType, filename];
}
